syntax = "proto3";

option java_multiple_files = false;
option java_package = "st.backend.stocks";
option java_outer_classname = "StockServiceProto";

service StockService {
  rpc GetStock1 (StockRequest1) returns (StockResponse1);
  rpc GetStock2 (StockRequest2) returns (StockResponse2);
}

message StockRequest1 {
  string id = 1;
  int64 start_time = 2;
  int64 end_time = 3;
}

message StockResponse1 {
  repeated StockResponseData data = 1;
  string message = 2;
}

message StockRequest2 {
  string name = 1;
  int64 start_time = 2;
  int64 end_time = 3;
}

message StockResponse2 {
  repeated StockResponseData data = 1;
  string message = 2;
}

message StockResponseData {
  string id = 1;
  string price = 2;
  string exchange = 3;
  string turnover = 4;
  string volume = 5;
  string amplitude = 6;
  string name = 7;
  string highest = 8;
  string lowest = 9;
  int64 beijingTime = 10;
  string timestamp = 11;
}
